### ISSUE TITLE
Possible exploiting of delegation + extended payout window

### ISSUE THREAD

**Issue by gropox** [GolosChain/golos#308]
It could be possible to exploit delegation of GP + extended payout window.

1. you write a post
2. vote for the post
3. delegate GP somebody
4. somebody votes for the post
5 you withdraw GP and pay for extension of payout window for the post
6. after 1 week, you delegate GP to somebody2 and he votes for the post.
and so on

**Comment by gropox** [comment]
If I see it right, the costs for extension of cachout window, are lower a net_shares is higher

![image](https://user-images.githubusercontent.com/26039472/33244565-4f45c08a-d2fa-11e7-8cf0-e83e93f65894.png)

**Comment by Nemo1369** [comment]
@gropox Alright, according to #91 seems like a payout extension cost ```net_rshares``` value dependency is not required in the moment.

Releasing ```comment_object.net_rshares``` value just after payout window extension been performed eliminates the necessity of ```comment_payout_extension``` operation. I would need to perform some tests.

**Comment by Nemo1369** [comment]
@gropox Do you understand you can do the same stuff with delegation with only one payout window?

**Comment by gropox** [comment]
Yes, but only once. With extended payout window you can do it at least twice. For how long can be payout window be extended and how often? One GBG for one day without limit or there is a limit?

**Comment by Nemo1369** [comment]
@gropox What is the difference, to delegate your vesting a billion times and vote in one week, or delegate it a billion times every window? Well, in fact, we can just reset ```comment_object.net_rshares``` value every time the new window starts, and the advance of payout window extension would become about not wasting bandwidth.

**Comment by gropox** [comment]
You cannot delegate GP more then once in one week. The GP withdrawal for delegator takes one week, in order to avoid exploiting multiple votes with same GP. 

Without extended payout window

1. You write a post
2. Upvote it self
3. Delegate GP to acc2
4. acc2 upvotes the post
5. You start GP withdraw
you get your power back only in one week (7 days, ) after payout. 

https://github.com/GolosChain/golos/blob/master/libraries/chain/evaluators/steem_evaluator.cpp#L1587

**Comment by gropox** [comment]
At least in theory, whether it works like in on steemit, i didn't tested on testnet. I will test it today or tomorrow.

**Comment by Nemo1369** [comment]
@gropox Oops, I've rechecked the protocol config header. My fault, totally forgot I've changed vesting delegation values for testing purposes in a local repository. Rebuilt the test network daemon with one week value.

**Comment by Nemo1369** [comment]
@gropox Alright, cyclic delegation is barely resolved - now effective vesting shares do not contain vesting shares delegated from the post/comment author. According to an unavailability of redelegation, that should be enough.

**Comment by gropox** [comment]
But you can still exploit extended period. Because you can vote for a post of your friend. It does not matter, who is the author. The only one possibility i see to solve the problem, is to track owner of each vests assigned for a comment, but it is to complicated to implement and will consume too much memory. I guess.

Most of delegates are again this feature in the coming HF. 

https://docs.google.com/spreadsheets/d/1wNUfdGRCtnosNamtD159Qwsik3_DZ8fv8jbguu1zSg4/edit?usp=sharing

I would propose to postpone extended payout window to the next releases, if all side effects are considered.

**Comment by Nemo1369** [comment]
@gropox Alright. Tracking each vest does not look like a solution too - that would be the same as spending vests. Postponing for now with 774898340d
